semantic_model_generator/generate_model.py

```diff
@@ -146,6 +147,18 @@ def _raw_table_to_semantic_context_table(
             f"No valid columns found for table {raw_table.name}. Please verify that this table contains column's datatypes not in {OBJECT_DATATYPES}."
         )
 
+    primary_key = None
+    if allow_joins:
+        # Populate the primary key field if we were able to retrieve one during raw table construction.
+        # If not, leave a placeholder for the user to fill out.
+        primary_key = semantic_model_pb2.PrimaryKey(
+            columns=(
+                raw_table.primary_key
+                if raw_table.primary_key
+                else [_PLACEHOLDER_COMMENT]
```

Comment:
how does the placeholder look like? do we get the "primary_key" section, and then it's blank under it? or do we get 1 line of an empty string column as part of the pk?

Reply:
We get an empty string column, like so:

```
    base_table:
      database: CATRANSLATOR
      schema: ANALYTICS
      table: CUSTOMERS
    primary_key:
      columns:
        - '  ' # <FILL-OUT>
```

Reply:
ok i see.
we currently have this check on whether the column is valid:
https://github.com/snowflakedb/cortex/blob/master/snowpilot/semantic_context/protos/schema.py#L286
so the empty comment will throw an error.

technically primary keys are only required for tables that are part of join relationships. But guess it is ok to assume that if someone is defining a schema with some relationships, that all tables will be involved in some join (thus requiring a primary key).


think what you have now is good enough to start

Reply:
sounds good - I figured it'd be simpler to start off by populating primary keys for all tables since this info should be static. If we do get feedback that users are only specifying joins on subsets tables in the model, I'll look into moving this to the join editor step instead of on creation.

Reply:
actually, I think it's important for the generated model to work out of the box, and as-is this would break. I'll move this later in the flow and only do it for tables specified in joins